src/zepben/evolve/database/sqlite/writers/base_cim_writer.py

```diff
@@ -0,0 +1,104 @@
+#  Copyright 2021 Zeppelin Bend Pty Ltd
+#
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+from __future__ import annotations
+
+import logging
+from sqlite3 import Cursor
+from typing import Set, Optional
+
+from dataclassy import dataclass
+
+from zepben.evolve import NameType, Name
+from zepben.evolve.database.sqlite.tables.database_tables import DatabaseTables, PreparedStatement
+from zepben.evolve.database.sqlite.tables.iec61968.common_tables import TableDocuments, TableOrganisations, TableOrganisationRoles
+from zepben.evolve.database.sqlite.tables.iec61970.base.core_tables import TableIdentifiedObjects, TableNameTypes, TableNames
+from zepben.evolve.database.sqlite.writers.utils import try_execute_single_update
+from zepben.evolve.model.cim.iec61968.common.document import Document
+from zepben.evolve.model.cim.iec61968.common.organisation import Organisation
+from zepben.evolve.model.cim.iec61968.common.organisation_role import OrganisationRole
+from zepben.evolve.model.cim.iec61970.base.core.identified_object import IdentifiedObject
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["BaseCIMWriter"]
+
+
+@dataclass(slots=True)
+class BaseCIMWriter(object):
+    """
+    Helper methods for building insert statements. Note all fields need to be populated.
+    """
+
+    database_tables: DatabaseTables
+    cursor: Cursor
+    _failed_ids: Set[str] = set()
+
+    def _save_document(self, table: TableDocuments, insert: PreparedStatement, document: Document, description: str) -> bool:
+        insert.add_value(table.title.query_index, document.title)
+        # TODO: JVM seems to use Z as TZ offset (for UTC+0?) while python uses +HH:mm format. Need to investigate here
```

Comment:
can remove this TODO. also we're getting test failures in the CI I believe from here. Possibly related to test data using the wrong format? check the test errors I guess (or try run them locally)